src/Network/Wai/Log/Internal.hs

```diff
@@ -5,26 +5,30 @@ import Data.Aeson.Types (Value(..), object)
 import Data.ByteString.Builder (Builder)
 import Data.Text (Text)
 import Data.Time.Clock (UTCTime, diffUTCTime, getCurrentTime)
-import Data.UUID (UUID)
+import Data.UUID (toText)
 import Data.UUID.V4 (nextRandom)
 import Log (LogLevel)
-import Network.Wai (Application, responseToStream)
+import Network.Wai (Application, requestHeaders, responseToStream)
+import qualified Data.Text.Encoding as Text
 
-import Network.Wai.Log.Options (Options(..), ResponseTime(..), requestUUID)
+import Network.Wai.Log.Options (Options(..), ResponseTime(..), requestID)
 
 -- | This type matches the one returned by 'getLoggerIO'
 type LoggerIO = UTCTime -> LogLevel -> Text -> Value -> IO ()
 
--- | Create a logging 'Middleware' that takes request UUID
+-- | Create a logging 'Middleware' that takes request id
 -- given a 'LoggerIO' logging function and 'Options'
-logRequestsWith :: LoggerIO -> Options -> (UUID -> Application) -> Application
+logRequestsWith :: LoggerIO -> Options -> (Text -> Application) -> Application
 logRequestsWith loggerIO Options{..} mkApp req respond = do
-  uuid <- nextRandom
-  logIO "Request received" $ logRequest uuid req
+  -- If the `request_id` can't be found in the header generated one instead
+  reqID <- case lookup "X-Request-ID" (requestHeaders req) of
```

Comment:
"X-Request-ID" does not seem to be a standard header field.  Maybe it would make sense to leave the task of determining the ID for a particular request to the application using the library?

One approach could be to add `Maybe (Request -> Id)` to `Options`, and when it's `Just` the library would use it to retrieve the request ID instead of generating UUID.

Or we could change the type of `mkLogMiddleware` to sth. like:
```
mkLogMiddleware :: (MonadLog m) => (Request -> IO Id) -> m LogMiddleware

Reply:
> "X-Request-ID" does not seem to be a standard header field. Maybe it would make sense to leave the task of determining the ID for a particular request to the application using the library?

If by "standard" you mean "it is standardised by the RFCs", no indeed. But it is common enough in practice that it deserves an entry on the Wikipedia page for HTTP Headesr: https://en.wikipedia.org/wiki/List_of_HTTP_header_fields#Common_non-standard_response_fields

Even the [Nginx website](https://www.nginx.com/blog/application-tracing-nginx-plus/) mentions it, so it's not a bad choice.

Reply:
Yes but according to Wikipedia it's not the only one used for this purpose, "X-Correlation-ID" and "Correlation-ID" are supposedly other common options.  And since it's not an enforce standard, other applications might use yet another header.

By hard-coding "X-Request-ID" we make an implicit choice of supporting only one of the possible headers and I don't think this is a good choice.

Reply:
I agree with @kawu, there's no point in hard-coding the header when providing option to pass a function to extract id from the `Request` is fairly simple and much more flexible.